packages/blade/src/utils/useMergeRefs.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { useMemo } from 'react';
+
+type ReactRef<T> = React.RefCallback<T> | React.MutableRefObject<T>;
+
+function assignRef<T = any>(ref: ReactRef<T> | null | undefined, value: T) {
+  if (ref == null) return;
+
+  if (typeof ref === 'function') {
+    ref(value);
+    return;
+  }
+
+  try {
+    ref.current = value;
```

Comment:
are we assigning ref.current(incoming ref.current) to internal ref? basically, does it mean ref.current=internalref? if yes then how will focus of internal ref work?